test/unit-tests/testAnalyticsFramework.js

```diff
@@ -569,19 +570,79 @@ describe('Analytics Framework Unit Tests', function()
     var badParamsHelper = function(framework, params, msgSentObj)
     {
       var msgName;
-      for(msgName in OO.Analytics.EVENTS)
+      var events = framework.flattenEvents(OO.Analytics.EVENTS); 
+      for(var i = 0; i < events.length; i++)
       {
-        expect(framework.publishEvent(OO.Analytics.EVENTS[msgName])).toBe(true);
+        msgName = events[i];
+        expect(framework.publishEvent(msgName)).toBe(true);
         msgSentObj.count++;
         recordedEvents = framework.getRecordedEvents();
         expect(_.isArray(recordedEvents)).toBe(true);
         var length = recordedEvents.length;
         expect(length).toEqual(msgSentObj.count);
         var lastMsg = recordedEvents[length-1];
-        expect(lastMsg.eventName).toEqual(OO.Analytics.EVENTS[msgName]);
+        expect(lastMsg.eventName).toEqual(msgName);
```

Comment:
I take that back. If you add to the flatten unit test, below, then it will cover what I was worried about.